### ISSUE TITLE
Protobuff

### ISSUE THREAD

**Issue by brothercorvo** [FreeTAKTeam/FreeTakServer#243]

**Comment by brothercorvo** [comment]
related to services @naman108 ?

**Comment by naman108** [comment]
I don't think this should be closed as we still want to implement protobuf @brothercorvo

**Comment by naman108** [comment]
This issue has several dimensions. First of all, protobuf in python requires the generation of protobuf domain files, this would be a useful application of the generation work. Secondly protobuf would probably require the implementation of a new serialization controller in the DigitalPy serialization component. Finally theres the question of latency as we found in past experiments that the python implementation protobuf is **extremely** slow.

**Comment by brothercorvo** [comment]
Python speed is a concern but the real conceptual issue here is that switching protocol over the same port is an **anti pattern**, typically you would go to a completely different port. If we implement that we break our network/service implementation

**Comment by naman108** [comment]
> Python speed is a concern but the real conceptual issue here is that switching protocol over the same port is an **anti pattern**, typically you would go to a completely different port. If we implement that we break our network/service implementation

Yes, that is another issue. Though neither the network, nor the service is responsible for the serialization process.
